src/promptflow-evals/tests/evals/e2etests/README.md

```diff
@@ -0,0 +1,37 @@
+# E2E tests logging to the application insights.
+For the regression tracking we are recording test run times and estimate the installation times of promptflow-evals wheel with and without `azure` optional dependency. The table below summarizes the events we are logging.
+
+## Events
+| Event | Description | Logging value | Is recorded |
+|-------|-------------|---------------|-------------|
+| install_promptflow_no_extras | Installation time of the packages without Azure | The number of packages installed and installation time in seconds | N/A |
+| install_promptflow_with_extras | Installation time of the packages with Azure | The number of packages installed and installation time in seconds | N/A |
+| run_e2e_tests_local | Run time tests marked as `localtest`\*. Each test is being logged as `TestClass::test_name`. | The test run time in seconds | No |
+| run_e2e_tests_azure | Run time tests marked as `azuretest`\*. Each test is being logged as `TestClass::test_name`. These tests write tracking information to Azure. | The test run time in seconds | Yes |
+| performance_tests | Run two tests parked as `performance_test` Each test is being logged as `TestClass::test_name`. These tests write tracking information to Azure. | The test run time in seconds | No |
+
+\* `localtest` is used to mark the whole class, while `azuretest` can be used to only mark tests that need `promptflow-azure` package.
+
+## Sample query
+The next query demonstrates the selection of installation times of promptflow-evals without additional dependencies for time after 2024-07-02 and plotting averges for different dates and operation systems.
+
+```
+let date_of_run=datetime("2024-07-02");
+let event='install_promptflow_no_extras';
+customEvents
+| where timestamp > date_of_run
+| where name == event
+| extend parsed = parse_json(customDimensions)
+| project 
+      name,
+      run_date=format_datetime(timestamp, 'yyyy-MM-dd'),
+      metric = parsed['activity_name'],
+      OS = strcat(parsed["OS"], "_", parsed["OS_release"]),
+      python_version = tostring(parsed['python_version']),
+      install_time_s=todecimal(parsed["install_time_s"])
+| summarize install_time_s = avg(install_time_s) by run_date, OS //python_version, stdev=stdev(value)

```

Comment:
Is there a way we can distinguish between the PR and scheduled run?

Also, I think we should make some adjustments to the pipelines

- For PRs, run all types of tests (local, Azure, and perf) in recordings mode. (For saving tokens)
- For pipeline scheduled runs, run the tests in live mode. (For more accurate metrics)

Reply:
We can check, which branch we are running on and set the recording value appropriately, however, in this case we will, essentially run live all the time, because the PR will always result in new recording and running live will not use the recording.

Reply:
We can uncomment below line in pipeline file. And send github.event_name to app insights if we haven't done so.

`#run: echo "PROMPT_FLOW_TEST_MODE=$(if [[ "${{ github.event_name }}" == "pull_request" ]]; then echo replay; else echo live; fi)" >> $GITHUB_ENV`